R/github-labels.R

```diff
@@ -25,51 +44,51 @@ use_github_labels <- function(delete_default = FALSE,
     )
   }
 
-  labels <- gh("GET /repos/:owner/:repo/labels")
+  cur_labels <- gh("GET /repos/:owner/:repo/labels")
 
   # Add missing labels
-  if (identical(labels[[1]], "")) {
-    cur_labels <- character()
+  if (identical(cur_labels[[1]], "")) {
+    cur_label_names <- character()
   } else {
-    cur_labels <- vapply(labels, "[[", "name", FUN.VALUE = character(1))
+    cur_label_names <- vapply(cur_labels, "[[", "name", FUN.VALUE = character(1))
   }
-  new_labels <- setdiff(names(gh_labels), cur_labels)
+  new_labels <- setdiff(names(labels), cur_label_names)
   if (length(new_labels) > 0) {
     done("Adding missing labels: ", collapse(value(new_labels)))
 
     for (label in new_labels) {
       gh(
         "POST /repos/:owner/:repo/labels",
         name = label,
-        color = gh_labels[[label]]
+        color = labels[[label]]
       )
     }
   }
 
   # Correct bad colours
-  if (identical(labels[[1]], "")) {
+  if (identical(cur_labels[[1]], "")) {
     cur_cols <- character()
   } else {
-    cur_cols <- vapply(labels, "[[", "color", FUN.VALUE = character(1))
+    cur_cols <- vapply(cur_labels, "[[", "color", FUN.VALUE = character(1))
   }
-  tru_cols <- gh_labels[cur_labels]
-  col_labels <- cur_labels[!is.na(tru_cols) & tru_cols != cur_cols]
+  tru_cols <- labels[cur_label_names]
+  col_labels <- cur_label_names[!is.na(tru_cols) & tru_cols != cur_cols]
 
-  if (length(col_labels) > 1) {
+  if (length(col_labels) > 0) {
     done("Setting label colours: ", collapse(value(col_labels)))
 
     for (label in col_labels) {
       gh(
         "PATCH /repos/:owner/:repo/labels/:name",
         name = label,
-        color = gh_labels[[label]]
+        color = labels[[label]]
       )
     }
   }
 
-  if (delete_default) {
-    default <- vapply(labels, "[[", "default", FUN.VALUE = logical(1))
-    def_labels <- setdiff(cur_labels[default], names(gh_labels))
+  if (delete_default && length(cur_labels) > 0) {
+    default <- vapply(cur_labels, "[[", "default", FUN.VALUE = logical(1))
+    def_labels <- setdiff(cur_label_names[default], names(labels))
 
     if (length(def_labels) > 0) {
       done("Removing default labels: ", collapse(value(def_labels)))
```

Comment:
default -> other ?

Reply:
Now reads "Removing labels", which should stand the test of time.